src/components/Table/BaseTable.tsx

```diff
@@ -51,127 +55,171 @@ type SpruceTableProps = {
-      <TableHead>
-        {table.getHeaderGroups().map((headerGroup) => (
-          <HeaderRow key={headerGroup.id}>
-            {headerGroup.headers.map((header) => {
-              const { columnDef } = header.column ?? {};
-              const { meta } = columnDef;
-              return (
-                <HeaderCell
-                  key={header.id}
-                  header={header}
-                  style={meta?.width && { width: columnDef?.meta?.width }}
-                >
-                  {flexRender(columnDef.header, header.getContext())}
-                  {header.column.getCanFilter() &&
-                    (meta?.treeSelect ? (
-                      <TableFilterPopover
-                        data-cy={meta.treeSelect?.["data-cy"]}
-                        onConfirm={(value) =>
-                          header.column.setFilterValue(value)
-                        }
-                        options={
-                          meta.treeSelect?.filterOptions
-                            ? meta.treeSelect.options.filter(
-                                ({ value }) =>
-                                  !!header.column
-                                    .getFacetedUniqueValues()
-                                    .get(value)
-                              )
-                            : meta.treeSelect.options
-                        }
-                        value={
-                          (header?.column?.getFilterValue() as string[]) ?? []
-                        }
-                      />
-                    ) : (
-                      <TableSearchPopover
-                        data-cy={meta?.search?.["data-cy"]}
-                        onConfirm={(value) =>
-                          header.column.setFilterValue(value)
-                        }
-                        placeholder={meta?.search?.placeholder}
-                        value={
-                          (header?.column?.getFilterValue() as string) ?? ""
-                        }
-                      />
-                    ))}
-                </HeaderCell>
-              );
-            })}
-          </HeaderRow>
-        ))}
-      </TableHead>
-      <TableBody>
-        {loading && (
-          <TableLoader
-            numColumns={table.getAllColumns().length}
-            numRows={loadingRows}
-          />
-        )}
-        {table.getRowModel().rows.map((row) => (
-          <Row
-            key={row.id}
-            row={row}
-            data-cy="leafygreen-table-row"
-            className={css`
-              &[aria-hidden="false"] td > div {
-                max-height: unset;
-              }
-            `}
-          >
-            {row.getVisibleCells().map((cell) => (
-              <Cell key={cell.id} style={{ padding: `${size.xxs} 2px` }}>
-                {flexRender(cell.column.columnDef.cell, cell.getContext())}
-              </Cell>
+export const BaseTable = forwardRef(
```

Comment:
I believe the typing issues we discussed came from an inability to use generics with `forwardRef`. I think removing the `forwardRef` is the cleanest option, but if you see a strong need for it, I have some more ideas from [this article](https://fettblog.eu/typescript-react-generic-forward-refs/) that we could discuss!

Reply:
I responded above on why forwardRef is required. Do you have any preference on which of those implementations we should use?

Reply:
I think Option 2 with a wrapper component seems pretty straightforward. But whatever you like works for me!

Reply:
Discussed this offline. It doesn't really make sense to type this with generics since this component is meant to be generic and reusable and is agnostic of any data type passed into it.